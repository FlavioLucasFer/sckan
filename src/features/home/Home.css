@media screen and (max-width: 1440px) {
	.action-card {
		padding: 16px;
	}
	
	.action-icon {
		font-size: 40px;
	}
}

@media screen and (max-width: 1200px) {
	.action-card {
		padding: 14px;
	}
	
	.action-icon {
		font-size: 35px;
	}
}

@media screen and (max-width: 992px) {
	.action-card {
		padding: 12px !important;
	}
	
	.action-icon {
		font-size: 30px !important;
	}
}

@media screen and (max-width: 600px) {
	.action-card {
		padding: 24px !important;
	}
	
	.action-icon {
		font-size: 60px !important;
	}
	
}