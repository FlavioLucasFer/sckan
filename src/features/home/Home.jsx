import './Home.css';
import React from 'react';
import { useHistory } from 'react-router-dom';

import FeatureCard from './components/FeatureCard';
import CardPanel from 'core/components/CardPanel';
import Icon from 'core/components/Icon';
import { COMPANY_FORM_ROUTE, COMPANY_ROUTE, PROJECT_FORM_ROUTE, PROJECT_ROUTE, SPRINT_FORM_ROUTE, SPRINT_ROUTE, TASK_FORM_ROUTE, TASK_ROUTE, USER_FORM_ROUTE, USER_ROUTE } from 'core/utils/routes';

function Home() {
	const { push } = useHistory();

	const userName = 'Flavio'

	return (
		<div>
			<h2 className='screen-title grey-text text-darken-3'>Welcome back, {userName}!</h2>

			<section>
				<h5>What do you want to do?</h5>

				<div className="row">
					<div className="col s12 m6 xl4">
						<FeatureCard title="Tasks"
							featureIcon="task">
							<CardPanel cardClassName="waves-effect waves-light action-card"
								backgroundColor="green accent-4"
								tooltip="Add new task"
								onClick={() => push(TASK_FORM_ROUTE)}>
								<Icon className="action-icon"
									size="medium"
									color="white-text">
									add
								</Icon>
							</CardPanel>

							<CardPanel cardClassName="waves-effect waves-light action-card"
								backgroundColor="light-blue"
								tooltip="Tasks list"
								onClick={() => push(TASK_ROUTE)}>
								<Icon className="action-icon"
									size="medium"
									color="white-text">
									list
								</Icon>
							</CardPanel>
						</FeatureCard>
					</div>

					<div className="col s12 m6 xl4">
						<FeatureCard title="Users"
							featureIcon="manage_accounts">
							<CardPanel cardClassName="waves-effect waves-light action-card"
								backgroundColor="green accent-4"
								tooltip="Add new user"
								onClick={() => push(USER_FORM_ROUTE)}>
								<Icon className="action-icon"
									size="medium"
									color="white-text">
									add
								</Icon>
							</CardPanel>

							<CardPanel cardClassName="waves-effect waves-light action-card"
								backgroundColor="light-blue"
								tooltip="Users list"
								onClick={() => push(USER_ROUTE)}>
								<Icon className="action-icon"
									size="medium"
									color="white-text">
									list
								</Icon>
							</CardPanel>
						</FeatureCard>
					</div>

					<div className="col s12 m6 xl4">
						<FeatureCard title="Companies"
							featureIcon="business">
							<CardPanel cardClassName="waves-effect waves-light action-card"
								backgroundColor="green accent-4"
								tooltip="Add new company"
								onClick={() => push(COMPANY_FORM_ROUTE)}>
								<Icon className="action-icon" 
									size="medium"
									color="white-text">
									add
								</Icon>
							</CardPanel>

							<CardPanel cardClassName="waves-effect waves-light action-card"
								backgroundColor="light-blue"
								tooltip="Companies list"
								onClick={() => push(COMPANY_ROUTE)}>
								<Icon className="action-icon" 
									size="medium"
									color="white-text">
									list
								</Icon>
							</CardPanel>
						</FeatureCard>
					</div>
					
					<div className="col s12 m6 xl4">
						<FeatureCard title="Sprints"
							featureIcon="wysiwyg">
							<CardPanel cardClassName="waves-effect waves-light action-card"
								backgroundColor="green accent-4"
								tooltip="Add new sprint"
								onClick={() => push(SPRINT_FORM_ROUTE)}>
								<Icon className="action-icon" 
									size="medium"
									color="white-text">
									add
								</Icon>
							</CardPanel>

							<CardPanel cardClassName="waves-effect waves-light action-card"
								backgroundColor="light-blue"
								tooltip="Sprints list"
								onClick={() => push(SPRINT_ROUTE)}>
								<Icon className="action-icon" 
									size="medium"
									color="white-text">
									list
								</Icon>
							</CardPanel>
						</FeatureCard>
					</div>

					<div className="col s12 m6 xl4">
						<FeatureCard title="Projects"
							featureIcon="source">
							<CardPanel cardClassName="waves-effect waves-light action-card"
								backgroundColor="green accent-4"
								tooltip="Add new project"
								onClick={() => push(PROJECT_FORM_ROUTE)}>
								<Icon className="action-icon"
									size="medium"
									color="white-text">
									add
								</Icon>
							</CardPanel>

							<CardPanel cardClassName="waves-effect waves-light action-card"
								backgroundColor="light-blue"
								tooltip="Projects list"
								onClick={() => push(PROJECT_ROUTE)}>
								<Icon className="action-icon"
									size="medium"
									color="white-text">
									list
								</Icon>
							</CardPanel>
						</FeatureCard>
					</div>

					<CardPanel cardClassName="s12 m2 waves-effect waves-light"
						backgroundColor="grey lighten-4"
						tooltip="Settings">
						<Icon type="round" 
							size="large"
							color="grey-text text-darken-3">
							settings
						</Icon>
					</CardPanel>
				</div>
			</section>
		</div>
	);
}

export default Home;