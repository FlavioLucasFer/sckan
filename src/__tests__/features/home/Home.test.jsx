import React from 'react';
import { shallow } from 'enzyme';

import Home from 'features/home/Home';
import { 
	COMPANY_FORM_ROUTE, 
	COMPANY_ROUTE, 
	PROJECT_FORM_ROUTE, 
	PROJECT_ROUTE, 
	SPRINT_FORM_ROUTE, 
	SPRINT_ROUTE, 
	TASK_FORM_ROUTE, 
	TASK_ROUTE, 
	USER_FORM_ROUTE, 
	USER_ROUTE,
	USER_SITTINGS,
} from 'core/utils/routes';

const mockHistory = {
	push: jest.fn(),
};

jest.mock('react-router-dom', () => ({
	...(jest.requireActual('react-router-dom')),
	useHistory: () => (mockHistory),
}));

describe('Testing Home screen component', () => {
	function testCardNavigation(cardId, route) {
		const wrapper = shallow(
			<Home />
		);

		wrapper.find({ id: cardId })
			.dive()
			.find('.card-panel')
			.simulate('click');

		expect(mockHistory.push).toBeCalledWith(route);

	}
	
	afterAll(() => {
		jest.clearAllMocks();
	});
	
	it('should render correctly', () => {
		const wrapper = shallow(
			<Home />
		);

		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('should navigate to task form screen when click on add task card', () => {
		testCardNavigation('add-task-card', TASK_FORM_ROUTE);
	});

	it('should navigate to tasks list screen when click on list tasks card', () => {
		testCardNavigation('tasks-list-card', TASK_ROUTE);
	});

	it('should navigate to user form screen when click on add user card', () => {
		testCardNavigation('add-user-card', USER_FORM_ROUTE);
	});

	it('should navigate to users list screen when click on list users card', () => {
		testCardNavigation('users-list-card', USER_ROUTE);
	});

	it('should navigate to company form screen when click on add company card', () => {
		testCardNavigation('add-company-card', COMPANY_FORM_ROUTE);
	});

	it('should navigate to companies list screen when click on list companies card', () => {
		testCardNavigation('companies-list-card', COMPANY_ROUTE);
	});

	it('should navigate to sprint form screen when click on add sprint card', () => {
		testCardNavigation('add-sprint-card', SPRINT_FORM_ROUTE);
	});

	it('should navigate to sprints list screen when click on list sprints card', () => {
		testCardNavigation('sprints-list-card', SPRINT_ROUTE);
	});

	it('should navigate to project form screen when click on add project card', () => {
		testCardNavigation('add-project-card', PROJECT_FORM_ROUTE);
	});

	it('should navigate to projects list screen when click on list projects card', () => {
		testCardNavigation('projects-list-card', PROJECT_ROUTE);
	});

	it('should navigate to user settings screen when click on settings card', () => {
		testCardNavigation('user-settings-card', USER_SITTINGS);
	});
});