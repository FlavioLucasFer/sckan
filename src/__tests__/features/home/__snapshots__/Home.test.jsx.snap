// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home screen component should render correctly 1`] = `
"<div>
  <h2 className=\\"screen-title grey-text text-darken-3\\">
    Welcome back, 
    Flavio
    !
  </h2>
  <section>
    <h5>
      What do you want to do?
    </h5>
    <div className=\\"row\\">
      <div className=\\"col s12 m6 xl4\\">
        <FeatureCard title=\\"Tasks\\" featureIcon=\\"task\\">
          <CardPanel id=\\"add-task-card\\" cardClassName=\\"waves-effect waves-light action-card\\" backgroundColor=\\"green accent-4\\" tooltip=\\"Add new task\\" onClick={[Function: onClick]}>
            <Icon className=\\"action-icon\\" size=\\"medium\\" color=\\"white-text\\">
              add
            </Icon>
          </CardPanel>
          <CardPanel id=\\"tasks-list-card\\" cardClassName=\\"waves-effect waves-light action-card\\" backgroundColor=\\"light-blue\\" tooltip=\\"Tasks list\\" onClick={[Function: onClick]}>
            <Icon className=\\"action-icon\\" size=\\"medium\\" color=\\"white-text\\">
              list
            </Icon>
          </CardPanel>
        </FeatureCard>
      </div>
      <div className=\\"col s12 m6 xl4\\">
        <FeatureCard title=\\"Users\\" featureIcon=\\"manage_accounts\\">
          <CardPanel id=\\"add-user-card\\" cardClassName=\\"waves-effect waves-light action-card\\" backgroundColor=\\"green accent-4\\" tooltip=\\"Add new user\\" onClick={[Function: onClick]}>
            <Icon className=\\"action-icon\\" size=\\"medium\\" color=\\"white-text\\">
              add
            </Icon>
          </CardPanel>
          <CardPanel id=\\"users-list-card\\" cardClassName=\\"waves-effect waves-light action-card\\" backgroundColor=\\"light-blue\\" tooltip=\\"Users list\\" onClick={[Function: onClick]}>
            <Icon className=\\"action-icon\\" size=\\"medium\\" color=\\"white-text\\">
              list
            </Icon>
          </CardPanel>
        </FeatureCard>
      </div>
      <div className=\\"col s12 m6 xl4\\">
        <FeatureCard title=\\"Companies\\" featureIcon=\\"business\\">
          <CardPanel id=\\"add-company-card\\" cardClassName=\\"waves-effect waves-light action-card\\" backgroundColor=\\"green accent-4\\" tooltip=\\"Add new company\\" onClick={[Function: onClick]}>
            <Icon className=\\"action-icon\\" size=\\"medium\\" color=\\"white-text\\">
              add
            </Icon>
          </CardPanel>
          <CardPanel id=\\"companies-list-card\\" cardClassName=\\"waves-effect waves-light action-card\\" backgroundColor=\\"light-blue\\" tooltip=\\"Companies list\\" onClick={[Function: onClick]}>
            <Icon className=\\"action-icon\\" size=\\"medium\\" color=\\"white-text\\">
              list
            </Icon>
          </CardPanel>
        </FeatureCard>
      </div>
      <div className=\\"col s12 m6 xl4\\">
        <FeatureCard title=\\"Sprints\\" featureIcon=\\"wysiwyg\\">
          <CardPanel id=\\"add-sprint-card\\" cardClassName=\\"waves-effect waves-light action-card\\" backgroundColor=\\"green accent-4\\" tooltip=\\"Add new sprint\\" onClick={[Function: onClick]}>
            <Icon className=\\"action-icon\\" size=\\"medium\\" color=\\"white-text\\">
              add
            </Icon>
          </CardPanel>
          <CardPanel id=\\"sprints-list-card\\" cardClassName=\\"waves-effect waves-light action-card\\" backgroundColor=\\"light-blue\\" tooltip=\\"Sprints list\\" onClick={[Function: onClick]}>
            <Icon className=\\"action-icon\\" size=\\"medium\\" color=\\"white-text\\">
              list
            </Icon>
          </CardPanel>
        </FeatureCard>
      </div>
      <div className=\\"col s12 m6 xl4\\">
        <FeatureCard title=\\"Projects\\" featureIcon=\\"source\\">
          <CardPanel id=\\"add-project-card\\" cardClassName=\\"waves-effect waves-light action-card\\" backgroundColor=\\"green accent-4\\" tooltip=\\"Add new project\\" onClick={[Function: onClick]}>
            <Icon className=\\"action-icon\\" size=\\"medium\\" color=\\"white-text\\">
              add
            </Icon>
          </CardPanel>
          <CardPanel id=\\"projects-list-card\\" cardClassName=\\"waves-effect waves-light action-card\\" backgroundColor=\\"light-blue\\" tooltip=\\"Projects list\\" onClick={[Function: onClick]}>
            <Icon className=\\"action-icon\\" size=\\"medium\\" color=\\"white-text\\">
              list
            </Icon>
          </CardPanel>
        </FeatureCard>
      </div>
      <CardPanel id=\\"user-settings-card\\" cardClassName=\\"s12 m2 waves-effect waves-light\\" backgroundColor=\\"grey lighten-4\\" tooltip=\\"Settings\\" onClick={[Function: onClick]}>
        <Icon type=\\"round\\" size=\\"large\\" color=\\"grey-text text-darken-3\\">
          settings
        </Icon>
      </CardPanel>
    </div>
  </section>
</div>"
`;
